src/zcash/JoinSplit.cpp

```diff
@@ -293,11 +293,15 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
         // estimate that it doesn't matter if we check every time.
         pb.constraint_system.swap_AB_if_beneficial();
 
-        r1cs_ppzksnark_proving_key<ppzksnark_ppT> pk;
-        loadFromFile(pkPath, pk);
+        std::ifstream fh(pkPath, std::ios::binary);
 
-        return ZCProof(r1cs_ppzksnark_prover<ppzksnark_ppT>(
-            pk,
+        // TODO
+        if(!fh.is_open()) {
+            throw std::runtime_error((boost::format("could not load param file at %s") % pkPath).str());
```

Comment:
There are two uses of ``boost::format`` in this file, which are the only ones in the codebase apart from in src/test/rpc_wallet_tests.cpp .